clustering/__init__.py

```diff
@@ -10,3 +23,197 @@ def cluster(pd_rules: pd.DataFrame) -> dict:
     data = ~pd_rules.isin([False, 'False', None, 'None', 0])
     clusters = KMeans(n_clusters=5).fit_predict(data)
     return dict(zip(pd_rules['system_id'], clusters))
+
+
+def preprocess_rules_data(DF):
+	
+	# Remove ID and time Columns from training set
+	DF = DF.iloc[:,1:-2] 
+	# Convert any remaining textual data into a nan value (there are very few of these in the dataset)
+	DF = DF.apply(lambda L: pd.to_numeric(L,errors='coerce'))
+	# Fill in any remaning nan values with 0. 
+	DF = DF.fillna(0)
+	return DF
+
+
+def group_clusters(results,num_clusters):
+
+	day = {}
+
+	# initialize our day dictionary with our initial classes
+	for i in range(num_clusters):
+		current_list = []
+		for j in range(len(results)):
+			if i == results[j][1]:
+				current_list.append(results[j][0]) 
+		day[str(i)] = set(current_list)
+
+
+	return day
+
+
+def calculate_stability_score(day_1,day_2,num_clusters,ids_in_both_days):
```

Comment:
shouldn't we test that the systems are being put into the same clusters, rather than sizes of clusters?

Also, day1 can have 100k systems and day2 can have 200k systems total. So that would show as unstable?

Reply:
ah, I see, you do the actual system ids difference. So we should test this only on ids in both days?

Or is ```((1-(stability_score/len(ids_in_both_days))) * 100.0 )``` supposed to prevent that some system ids are new and some missing messing up with the score?

Reply:
oh I see, you do clear it in https://github.com/RedHatInsights/aicoe-insights-clustering/pull/12/files#diff-2a440676ce12f1a1b39194070ef96a56R191 and https://github.com/RedHatInsights/aicoe-insights-clustering/pull/12/files#diff-2a440676ce12f1a1b39194070ef96a56R198

since those are the same, it would make sense to put it in the method and call it here on both data sets